crates/datasources/src/sqlite/async_sqlite.rs

```diff
@@ -0,0 +1,232 @@
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use datafusion::arrow::datatypes::DataType;
+use futures::Stream;
+use rusqlite::types::Value;
+use rusqlite::{Connection, Params};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TrySendError;
+use tokio::task::JoinHandle;
+
+use crate::sqlite::errors::{Result, SqliteError};
+
+#[derive(Clone, Debug)]
+pub struct SqliteAsyncClient {
+    path: PathBuf,
+}
+
+impl SqliteAsyncClient {
```

Comment:
This client is synchronous, though?

Reply:
More like "async friendly".

Reply:
seems like we could make it actually async, right?

Reply:
What do you mean by "actually" async? Sqlite has no concept of async, I don't think we can do much about it apart from coloring the function as "async"

Reply:
oh just that the block-in-place could maybe go away, but that only really matters if the consumers of this are async. I think this is fine as is.